native/exe/SageLauncher/launcher.cpp

```diff
@@ -368,15 +383,17 @@ void doStageCleaning(BOOL bService, BOOL bClient)
 	}
 
 	// Handle the deletions files first
-	FILE* fp = fopen("stageddeletes.txt", "r");
+	FILE* fp;
+	fopen_s(&fp, "stageddeletes.txt", "r");

```

Comment:
If you want to switch to fopen_s (not sure why you're changing that, guessing another MS warning) then you need to check the return value from the call. That's how it indicates failure, not by setting fp to NULL. Same for all the places you switched this below.